### PR TITLE
Add GenomicTest and GenomicShortVariant

### PR DESCRIPTION
**Pull request by rcdilorenzo**
The goal with these easy modules is to make them look as similar as possible to the FSS modules. Obviously, there will be some limitations, but we're taking as much advantage as possible.

### Example: Get short variants on chromosome 1 from all active tests

```python
import phc.easy as phc

phc.Project.set_current("BRCA")

# Genomic tests and the associated tests are extracted here
set_ids = phc.GenomicTest.get_data_frame(
    patient_id="8cb82aa0-7f2c-4fdb-bf91-0ed1b315392c", 
    status="ACTIVE",
    test_type="shortVariant",
    all_results=True,
).id.values.tolist()

# Pull first 1000 short variants on chr1
phc.GenomicShortVariant.get_data_frame(
    variant_set_ids=set_ids,
    chromosome=["chr1"],
    page_size=1000,
    log=True
).info()
```

Result:

```
{
    "url": "https://***.com/v1/genomics/variants",
    "method": "GET",
    "params": {
        "variantSetIds": "6b97a98b-17fd-49dd-a934-0e8a44af8bf9,7e89210b-db2f-4faa-bc38-e59d35c9f47d,2708a2b8-9422-4b1e-a0b0-69f176bccb9c,d0d90e66-2aec-43ca-b4d4-63df6d5a4dfa",
        "include": "vcf",
        "chromosome": "chr1",
        "pageSize": 1000
    }
}

Retrieved 1045/10000 results
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1045 entries, 0 to 1044
Data columns (total 31 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   id                           1045 non-null   object 
 1   chromosome                   1045 non-null   object 
 2   reference                    1045 non-null   object 
 3   alternate                    1045 non-null   object 
 4   position                     1045 non-null   int64  
 5   minimumAlleleFrequency       1045 non-null   float64
 6   maximumAlleleFrequency       1045 non-null   float64
 7   gnomadAlleleFrequency        1045 non-null   float64
 8   zygosity                     1045 non-null   object 
 9   clinvar.nearVariant          1045 non-null   int64  
 10  clinvar.alleleId             2 non-null      object 
 11  clinvar.disease              2 non-null      object 
 12  clinvar.review               2 non-null      object 
 13  clinvar.significance         2 non-null      object 
 14  clinvar.submission           2 non-null      object 
 15  cosmic.sampleCount           1045 non-null   int64  
 16  cosmic.nearVariant           1045 non-null   int64  
 17  cosmic.cosmicId              77 non-null     object 
 18  cosmic.status                77 non-null     object 
 19  cosmic.histology             77 non-null     object 
 20  cosmic.tumorSite             77 non-null     object 
 21  ensemblCanon                 980 non-null    object 
 22  dbnsfp                       1045 non-null   object 
 23  vcf.quality                  1045 non-null   float64
 24  vcf.filter                   1045 non-null   object 
 25  vcf.variantAllelicFrequency  1045 non-null   float64
 26  vcf.coverage                 1045 non-null   object 
 27  gnomadHomozygous             822 non-null    float64
 28  rsid                         932 non-null    object 
 29  exacAlleleFrequency          7 non-null      float64
 30  exacHomozygous               7 non-null      float64
dtypes: float64(8), int64(4), object(19)
memory usage: 253.2+ KB
```

### Note

The one thing I haven't done is that you previously have to nest arguments to the expansion (`Frame.expand`) under `expand_args=dict(code_columns=["identifier"])` in the FSS modules. With these new modules, you can directly add `code_columns` for example as an argument. That'll be something I retrofit to the FSS modules when I get a chance.

### CODE REVIEW

File: phc/easy/genomic_short_variant.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+from phc.easy.frame import Frame
+from pydantic import Field
+from phc.easy.auth import Auth
+from phc.easy.paging_api_item import PagingApiItem, PagingApiOptions
+
+from enum import Enum
+
+
+MAPPINGS = {
+    "variant_set_ids": "variantSetIds",
```
Comment (rcdilorenzo) [inline phc/easy/genomic_short_variant.py:13]: Hurrah for vim macros. 😄

File: phc/easy/genomic_short_variant.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    ):
+        """Execute a request for genomic short variants
+
+        ## Parameters
+
+        Query: `GenomicShortVariantOptions`
+
+        Execution: `phc.easy.query.Query.execute_paging_api`
+
+        Expansion: `phc.easy.frame.Frame.expand`
```
Comment (rcdilorenzo) [inline phc/easy/genomic_short_variant.py:211]: Each of these get auto-linked in the HTML documentation.

File: phc/easy/query/api_paging.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+def clean_params(params: dict):
+    return {
+        k: v
+        for k, v in params.items()
+        if ((v is not None) and (not isinstance(v, str) or len(v) > 0))
+    }
+
+
+def recursive_paging_api_call(
```
Comment (rcdilorenzo) [inline phc/easy/query/api_paging.py:20]: We want this to be somewhat parallel to the [recursive_execute_fhir_dsl(...) function](https://github.com/lifeomic/phc-sdk-py/blob/70e4da931b02e130952a3005245f4a13796f44ab/phc/easy/query/fhir_dsl.py#L82-L141).

File: requirements.txt [hunk 4]
```diff
@@ -6,4 +6,5 @@ pandas
 funcy
 lenses
 toolz
+pydantic
```
Comment (rcdilorenzo) [inline requirements.txt:9]: Pydantic gives us runtime validation and excellent error messages. 🎉

**Review by mschroering — APPROVED** [review]

**Review by RyanRatcliff — APPROVED** [review]

**Pull request merged**